internal/provider/resource_consul_cluster.go

```diff
@@ -629,9 +673,10 @@ func resourceConsulClusterUpdate(ctx context.Context, d *schema.ResourceData, me
 	// Confirm update fields have been changed
 	sizeChanged := d.HasChange("size")
 	versionChanged := d.HasChange("min_consul_version")
+	ipAllowlistChanged := d.HasChange("ip_allowlist")
```

Comment:
For any updates:
- Detect changes in `ip_allowlist`
- if changes, `buildIPAllowlist()` to validate there are no duplicates
- Send update request